src/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        public unsafe struct LinkLayerAddressInfo
+        {
+            public int InterfaceIndex;
+            public fixed byte AddressBytes[8];
+            public byte NumAddressBytes;
+            private byte __pading;
```

Comment:
> I would expect the compiler to properly determine padding and struct alignment for structs. From our experience, manual padding or struct packing is a source of AVs

We explicitly do not want the native compiler inserting padding automatically, as we want the shim structs to have the same layout (modulo pointer size) across architectures. As a result, most of our shim structs are manually padded out to 8-byte alignment when necessary and the compiler is configured to emit an error if it needs to manually insert padding.